src/CGIHandler.cpp

```diff
@@ -1,53 +1,47 @@
 #include "CGIHandler.hpp"
 
-CGIHandler::CGIHandler()
+CGIHandler::CGIHandler(EventManager &eventManager, Connection &connection)
+	: AResponseHandler(), _connection(connection), _eventManager(eventManager)
 {
 }
 
+// Copy Constructor
+CGIHandler::CGIHandler(const CGIHandler &other)
+	: AResponseHandler(other)
+	, _FDsRef(other._FDsRef)
+	, _pollFd(other._pollFd)
+	, _connection(other._connection)
+	, _eventManager(other._eventManager)
+{
+	// TODO: do we need deep copy here?
+}
+
+// Destructor
 CGIHandler::~CGIHandler()
 {
 }
 
-// CGIHandler::CGIHandler(Connection *conn)
-// {
-// 	_conn = conn;
-// }
-
-// CGIHandler &CGIHandler::operator=(const CGIHandler &other)
-// {
-// 	if (this != &other)
-// 	{
-// 		_conn = other._conn;
-// 		_FDsRef = other._FDsRef;
-// 		_pollFd = other._pollFd;
-// 	}
-// 	return *this;
-// }
+CGIHandler &CGIHandler::operator=(const CGIHandler &other)
+{
+	if (this != &other)
+	{
+		AResponseHandler::operator=(other);
+		_eventManager = other._eventManager;
+		_FDsRef = other._FDsRef;
+		_pollFd = other._pollFd;
+	}
+	return *this;
+}
 
 void CGIHandler::handleRequest(const HTTPRequest &request, HTTPResponse &response)
 {
-	CGIHandler cgiInstance;
-	// cgiInstance.setFDsRef(_FDsRef); // here we set the FDs to close later unused ones
 	MetaVariables env;
 	env.HTTPRequestToMetaVars(request, env);
-	// std::cout << env;
-	std::string cgiOutput = executeCGI(env);
-	// if cgioutput == "HTTP/1.1 500 Internal Server Error\r\nContent-Length: 0\r\n\r\n";
-	// then we should return 500 error
-	if (cgiOutput == "HTTP/1.1 500 Internal Server Error\r\nContent-Length: 0\r\n\r\n")
+	if (!executeCGI(env, response))
 	{
 		response.setStatusCode(500, "");
 		response.setBody("500 Internal Server Error");
```

Comment:
Do we go through Router after this? User may provide custom 500 error page

Reply:
After this we go through the router (but we skip it kind of). I wrote a line that if response.getIsCGI is skipping the part of the router that we already went thorough before accessing executeCGI. We already went through the Router before this and all the settings should be saved in Connection. I hope they are saved there.

Reply:
So this is not correct

Reply:
Why not? Normally in the previous code we were not going through the Router again after this. Now we are.